code/modules/client/preferences.dm

```diff
@@ -49,39 +49,23 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 
 	//character preferences
 	var/real_name						//our character's name
-	var/random_name = 0
-	var/random_body = 0
 	var/gender = MALE					//gender of character (well duh)
-	var/random_gender = 0
 	var/age = 30						//age of character
-	var/random_age = 0
 	var/underwear = "Nude"				//underwear type
-	var/random_underwear = 1
 	var/underwear_color = "000"			//underwear color
-	var/random_underwear_color = 1
 	var/undershirt = "Nude"				//undershirt type
-	var/random_undershirt = 1
 	var/socks = "Nude"					//socks type
-	var/random_socks = 1
 	var/backpack = DBACKPACK				//backpack type
-	var/random_backpack = 1
 	var/jumpsuit_style = PREF_SUIT		//suit/skirt
-	var/random_jumpsuit_style = 1
 	var/hairstyle = "Bald"				//Hair type
-	var/random_hairstyle = 1
 	var/hair_color = "000"				//Hair color
-	var/random_hair_color = 1
 	var/facial_hairstyle = "Shaved"	//Face hair type
-	var/random_facial_hairstyle = 1
 	var/facial_hair_color = "000"		//Facial hair color
-	var/random_facial_hair_color = 1
 	var/skin_tone = "caucasian1"		//Skin color
-	var/random_skin_tone = 1
 	var/eye_color = "000"				//Eye color
-	var/random_eye_color = 1
 	var/datum/species/pref_species = new /datum/species/human()	//Mutant race
-	var/random_species = 0
 	var/list/features = list("mcolor" = "FFF", "ethcolor" = "9c3030", "tail_lizard" = "Smooth", "tail_human" = "None", "snout" = "Round", "horns" = "None", "ears" = "None", "wings" = "None", "frills" = "None", "spines" = "None", "body_markings" = "None", "legs" = "Normal Legs", "moth_wings" = "Plain", "moth_markings" = "None")
+	var/list/randomise = list("name" = 0, "body" = 0, "species" = 0, "gender" = 0, "age" = 0, "underwear" = 1, "underwear_color" = 1, "undershirt" = 1, "socks" = 1, "backpack" = 1, "jumpsuit_style" = 1, "hairstyle" = 1, "hair_color" = 1, "facial_hairstyle" = 1, "facial_hair_color" = 1, "skin_tone" = 1, "eye_color" = 1)

```

Comment:
Use defines please.

Reply:
Do you mean FALSE and TRUE?

Reply:
Both true/false and the randomization type strings.